team-props/static-analysis/pmd-rules.xml

```diff
@@ -0,0 +1,142 @@
+  <exclude-pattern>.*Test.*</exclude-pattern>
+
+  <rule ref="rulesets/java/strictexception.xml">
+    <!-- Ignored because of the Callable pattern -->
+    <exclude name="SignatureDeclareThrowsException" />
+  </rule>
+  <rule ref="rulesets/java/typeresolution.xml">
+    <!-- Ignored because of the Callable pattern -->
+    <exclude name="SignatureDeclareThrowsException" />
+  </rule>
+
+  <rule ref="rulesets/java/naming.xml/ShortMethodName">
+    <properties>
+      <property name="xpath">
+        <value>
+          //MethodDeclarator[string-length(@Image) &lt; 2]
+        </value>
+      </property>
+    </properties>
+  </rule>
+
+  <rule ref="rulesets/java/braces.xml" />
+
+  <rule ref="rulesets/java/design.xml">
+    <exclude name="TooFewBranchesForASwitchStatement" />
+    <exclude name="EmptyMethodInAbstractClassShouldBeAbstract" />
+    <exclude name="FieldDeclarationsShouldBeAtStartOfClass" />
+    <exclude name="ConfusingTernary" />
+    <exclude name="AccessorMethodGeneration" />
+  </rule>
+
+  <rule ref="rulesets/java/design.xml/EmptyMethodInAbstractClassShouldBeAbstract">
+    <priority>5</priority>
+  </rule>
+
+  <!-- Suppress ImmutableField check for Api models -->
+  <rule ref="rulesets/java/design.xml/ImmutableField">
+    <properties>
+      <property name="violationSuppressXPath"
+        value="//ClassOrInterfaceDeclaration[starts-with(@Image, 'Api')]//FieldDeclaration" />
+    </properties>
+  </rule>
+
+  <rule ref="rulesets/java/unusedcode.xml" />
+  <rule ref="rulesets/java/logging-java.xml" />
+
+  <rule ref="rulesets/java/strings.xml" />
+
+  <!-- Suppress AvoidDuplicateLiterals check for annotations -->
+  <rule ref="rulesets/java/strings.xml/AvoidDuplicateLiterals">
+    <properties>
+      <property name="skipAnnotations" value="true" />
+    </properties>
+  </rule>
+
+  <rule ref="rulesets/java/migrating.xml" />
+  <rule ref="rulesets/java/optimizations.xml">
+    <exclude name="LocalVariableCouldBeFinal" />
+    <exclude name="MethodArgumentCouldBeFinal" />
+    <exclude name="AvoidInstantiatingObjectsInLoops" />
+  </rule>
+  <rule ref="rulesets/java/basic.xml" />
+  <rule ref="rulesets/java/sunsecure.xml" />
+  <rule ref="rulesets/java/coupling.xml">
+    <exclude name="LoosePackageCoupling" />
+    <!-- LawOfDemeter seems to be bugged -->
+    <exclude name="LawOfDemeter" />
+  </rule>
+
+  <rule ref="rulesets/java/imports.xml">
+    <exclude name="TooManyStaticImports" />
+  </rule>
+
+  <rule ref="rulesets/java/junit.xml" />
+  <rule ref="rulesets/java/naming.xml">
+    <exclude name="AbstractNaming" />
+    <exclude name="LongVariable" />
+    <exclude name="ShortVariable" />
+    <exclude name="ShortClassName" />
+    <exclude name="AvoidFieldNameMatchingMethodName" />
+  </rule>
+
+  <rule ref="rulesets/java/codesize.xml">
+    <exclude name="TooManyMethods" />
+    <exclude name="TooManyFields" />
+  </rule>
+
+  <!-- Suppress ExcessiveParameterList check for Api models ctors and factory methods -->
```

Comment:
Do we actually have any classes that start with `Api` in this project?

Reply:
🤔 I don't think so, we can probably kill this one.

Reply:
690c984